graphs/frequent_pattern_graph_miner.py

```diff
@@ -0,0 +1,183 @@
+'''
+FP-GraphMiner - A Fast Frequent Pattern Mining Algorithm for Network Graphs
+
+A novel Frequent Pattern Graph Mining algorithm,
+FP-GraphMiner, that compactly represents a set of network graphs as a Frequent Pattern Graph (or FP-Graph).
+This graph can be used to efficiently mine frequent subgraphs including maximal frequent subgraphs and maximum common subgraphs.
+
+URL:https://www.researchgate.net/publication/235255851_FP-GraphMiner_-_A_Fast_Frequent_Pattern_Mining_Algorithm_for_Network_Graphs
+
+'''
+def get_DE(EA):
+    '''
+    Return Distinct edges from edge array of multiple graphs
+    '''
+    DE=set()
+    for i in range(len(EA)):

```

Comment:
Using [__enumerate()__](https://docs.python.org/3/library/functions.html#enumerate) is always faster and more expressive than __range(len())__ unless you are modifying the variables inside the loop.